File: experiment/order_of_if/negative_first.rb
def greeting(user)
  if !user[:title]
    name = user[:first_name]
  else
    name = "#{user[:title]} #{user[:last_name]}"
  end

  return "Dear " + name
end

user = {
  title: "Mrs.",
  first_name: "Jane",
  last_name: "Marple"
}

result = greeting user

File: experiment/order_of_if/negative_first.py
def greeting(user):
  if not user.get('title'):
    name = user.get('first_name')
  else:
    name = user.get('title') + " " + user.get('last_name')

  return "Dear " + name

user = {
  'title': 'Miss',
  'first_name': 'Jane',
  'last_name': 'Marple'
}

result = greeting(user)